tests/test_remotesparkmagics.py

```diff
@@ -0,0 +1,94 @@
+from nose.tools import raises, with_setup, assert_equals
+from remotespark.RemoteSparkMagics import RemoteSparkMagics
+from mock import MagicMock
+
+ip = get_ipython()
+magic = None
+
+def _setup():
+	global magic
+	magic = RemoteSparkMagics(shell=ip)
+	ip.register_magics(magic)
+
+def _teardown():
+	pass
+
+@with_setup(_setup, _teardown)
+def test_info_command():
+	mock_method = MagicMock()
+	magic._print_info = mock_method
+	command = "info"
+
+	ip.run_line_magic("spark", command)
+
+	mock_method.assert_called_with()
+
+@with_setup(_setup, _teardown)
+def test_add_endpoint_command():
+	mock_method = MagicMock()
+	magic.add_endpoint = mock_method
+	command = "add"
+	name = "name"
+	language = "python"
+	connection_string = "url=http://location:port;username=name;password=word"
+	line = " ".join([command, name, language, connection_string])
+
+	ip.run_line_magic("spark", line)
+
+	mock_method.assert_called_with(name, language, connection_string)
+
+
+@with_setup(_setup, _teardown)
+def test_delete_endpoint_command():
+	mock_method = MagicMock()
+	magic.delete_endpoint = mock_method
+	command = "delete"
+	name = "name"
+	line = " ".join([command, name])
+
+	ip.run_line_magic("spark", line)
+
+	mock_method.assert_called_with(name)
```

Comment:
It adds a nicer level of granularity.  If we have separate parsing and functional tests, we can more easily identify where the problem is when one fails.  It also allows us to test the methods' functionality directly without all the magic invocation, which will become increasingly important when we have multiple ways to interact with the magic (command options vs. widgets).